specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts

```diff
@@ -251,12 +255,21 @@ export const reducer = generateReducer<MappingState, MappingActions>({
     )
       return state;
 
+    const newLine = replaceItem(state.lines, focusedLine, {
+      ...state.lines[focusedLine],
+      mappingPath: mappingViewMappingPath,
+    });
+
+    const newHeader = generateMappingPathPreview(
+      baseTableName as keyof Tables,
+      newLine[focusedLine].mappingPath
+    );
```

Comment:
hmm, doesn't look right
the `headerName` is supposed to be coming from the data set
the only time you would be setting header name to result of `generateMappingPathPreview()` is when you are trying to add a new column to the data set - which doesn't seem to be the case here?

Reply:
we need to reset the header when the user changes the fields in an existing line